docs/02.API-REFERENCE.md

```diff
@@ -8916,6 +8965,171 @@ best-practice example.
 - [jerry_object_native_info_t](#jerry_object_native_info_t)
 
 
+## jerry_native_pointer_init_references
+
+**Summary**
+
+Initialize the references stored in a buffer pointed by a native pointer.
+The references are initialized to undefined. This function must be called
+before the buffer is attached to an object by
+[jerry_set_object_native_pointer](#jerry_set_object_native_pointer).
+
+*Note*:
+  - The description of [jerry_object_native_info_t](#jerry_object_native_info_t)
+    provides detailed information about these references.
+
+**Prototype**
+
+```c
+void
+jerry_native_pointer_init_references (void *native_pointer_p,
+                                      const jerry_object_native_info_t *native_info_p);
+```
+
+- `native_pointer_p` - native pointer to a buffer.
+- `native_info_p` - native pointer's type information.
+
+*New in version [[NEXT_RELEASE]]*.
+
+**Example**
+
+[doctest]: # ()
+
+```c
+#include <stdlib.h>
+#include "jerryscript.h"
+
+typedef struct
+{
+  uint32_t user_data;
+  jerry_value_t a;
+  jerry_value_t b;
+  uint32_t user_other_data;
+} user_buffer_t;
+
+static void
+native_references_free_callback (void *native_p, /**< native pointer */
+                                 jerry_object_native_info_t *info_p) /**< native info */
+{
+  /* References must be finalized when a buffer is no longer attached. */
+  jerry_native_pointer_finalize_references (native_p, info_p);
+  free (native_p);
+} /* native_references_free_callback */
+
+static const jerry_object_native_info_t native_info =
+{
+  .free_cb = native_references_free_callback,
+  .number_of_references = JERRY_OBJECT_NATIVE_INFO_GET_NUMBER_OF_REFERENCES (user_buffer_t, a, b),
+  .offset_of_references = offsetof(user_buffer_t, a),
+};
+
+int
+main (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  jerry_value_t object_value = jerry_create_object ();
+
+  user_buffer_t *buffer_p = (user_buffer_t *) malloc (sizeof (user_buffer_t));
+
+  /* References must be initialized before a buffer is attached. */
+  jerry_native_pointer_init_references ((void *) buffer_p, &native_info);
+
+  jerry_set_object_native_pointer (object_value, (void *) buffer_p, &native_info);
+
+  /* References can be modified after the buffer is attached.
+   * This example sets a self reference. */
+  jerry_native_pointer_set_reference (&buffer_p->a, object_value);
+
+  jerry_release_value (object_value);
+
+  jerry_cleanup ();
+  return 0;
+}
+```
+
+**See also**
+
+- [jerry_set_object_native_pointer](#jerry_set_object_native_pointer)
+- [jerry_native_pointer_finalize_references](#jerry_native_pointer_finalize_references)
+- [jerry_native_pointer_set_reference](#jerry_native_pointer_set_reference)
+
+## jerry_native_pointer_release_references
```

Comment:
I thinks this is a typo as this should be `jerry_native_pointer_finalize_references` or the other places should use this name.